### PR TITLE
changes in username fields in the form

### PR DESCRIPTION
**Pull request by AmanelGau**

### CODE REVIEW

File: src/Service/UserService.php [hunk 1]
```diff
@@ -101,16 +85,11 @@ public function validateNewPassword(User $user, UserPasswordEncoderInterface $pa
...
     ): array {
         $errors = [];
-        if (!$isUsernameValid && $this->getOldUsername()) {
-            if ($this->getOldUsername() != $user->getUsername()) {
-                $errors[] = "Votre nom d'utilisateur est invalide";
-            } elseif (!$this->getNewUsername()) {
-                $errors[] = "Veuillez entré un nouveau nom d'utilisateur";
-            }
+        if (!$this->getNewUsername()) {
+            $errors[] = "Veuillez entré un nouveau nom d'utilisateur";
```
Comment (arnaud-blanc) [inline src/Service/UserService.php:92]: "entré" => "entrer"

**Review by arnaud-blanc — APPROVED** [review]
Stephane also did something interesting, if you want to ask about additional things!

File: src/Controller/SecurityController.php [hunk 2]
```diff
@@ -57,16 +57,17 @@ public function editUser(
             throw $this->createAccessDeniedException('vous devez etre connecté');
         }
         $userService = new UserService();
```
Comment (RomainClair) [inline src/Controller/SecurityController.php:59]: Objects behind forms are not services.
Is there a reason not to use the User object this forms seems to edit ?

File: src/Controller/SecurityController.php [hunk 3]
```diff
@@ -57,16 +57,17 @@ public function editUser(
...
             $isPasswordValid = $userService->validateNewPassword($user, $passwordEncoder);
 
-            $errors = $userService->findUserErrors($user, $passwordEncoder, $isUsernameValid, $isPasswordValid);
+            $errors = $userService->findUserErrors($user, $passwordEncoder, $isPasswordValid);
 
-            if ($isUsernameValid || $isPasswordValid) {
+            if (($isUsernameValid || $isPasswordValid) && !$errors) {
                 $entityManager->flush();
                 return $this->redirectToRoute('home');
             } elseif ($errors) {
```
Comment (RomainClair) [inline src/Controller/SecurityController.php:73]: It's a strange idea to bypass the Symfony Form component and it's validation tools in a symfony prooject.
You probably should use constraints in the User entity to validate your form edition.

**Review by RomainClair — CHANGES_REQUESTED** [review]
You probably should deal with form the symfony way (in a symfony project)
Take a look at : https://symfony.com/doc/current/forms.html

File: src/Controller/SecurityController.php [hunk 4]
```diff
@@ -56,25 +56,39 @@ public function editUser(
...
+
+                $this->addFlash('notice', 'Votre nom de compte à bien été changé !');
                 return $this->redirectToRoute('home');
-            } elseif ($errors) {
-                foreach ($errors as $error) {
-                    $this->addFlash('error', $error);
-                }
             } else {
-                $this->addFlash('warning', 'Champs manquant, rien n\'a été modifié');
+                $this->addFlash('notice', 'Aucune modification détecter, veuillez remplir les champs correctement !');
```
Comment (arnaud-blanc) [inline src/Controller/SecurityController.php:91]: détecter => détectée

**Review by arnaud-blanc — CHANGES_REQUESTED** [review]
Reminder: the administrator is not comfortable with computers!
It will be essential to clearly explain the steps to be taken to :
- change username
- change password
These are two different things.
For this, it is necessary to bring an explanatory note to the administrator.

**Review by Orijma — APPROVED** [review]

**Review by RomainClair — COMMENTED** [review]
Check the UserPassword constraints (https://symfony.com/doc/current/reference/constraints/UserPassword.html) which can be useful in this particular case
You should make sure to "forget" the plainPassword after encoding by completing the User::eraseCredential method and call it in you controller after encoding.

**Review by RomainClair — APPROVED** [review]
Check the UserPassword constraints (https://symfony.com/doc/current/reference/constraints/UserPassword.html) which can be useful in this particular case
You should make sure to "forget" the plainPassword after encoding by completing the User::eraseCredential method and call it in you controller after encoding.

**Review by RomainClair — DISMISSED** [review]

**Review by arnaud-blanc — APPROVED** [review]

**Review by Orijma — APPROVED** [review]

**Review by RomainClair — APPROVED** [review]

**Pull request merged**